core/src/test/java/com/novoda/noplayer/subtitles/SystemCaptionPreferencesTest.java

```diff
@@ -0,0 +1,289 @@
+            this.expectedTextSize = expectedTextSize;
+            return this;
+        }
+
+        TestCase expectedWindowColor(int expectedWindowColor) {
+            this.expectedWindowColor = expectedWindowColor;
+            return this;
+        }
+
+        TestCase expectedTypeface(Typeface expectedTypeface) {
+            this.expectedTypeface = expectedTypeface;
+            return this;
+        }
+
+        TestCase expectedEdgeType(int expectedEdgeType) {
+            this.expectedEdgeType = expectedEdgeType;
+            return this;
+        }
+
+        TestCase expectedEdgeColor(int expectedEdgeColor) {
+            this.expectedEdgeColor = expectedEdgeColor;
+            return this;
+        }
+
+        TestCase captionsManagerDisabled() {
+            this.captionsManagerIsEnabled = false;
+            return this;
+        }
+
+        TestCase captionsLocallyDisabled() {
+            this.captionsLocallyEnabled = false;
+            return this;
+        }
+
+        @NonNull
+        @Override
+        public String toString() {
+            return name;
+        }
+    }
+
+    private static SystemCaptionPreferences givenSystemCaptionPreferences(TestCase testCase) {
+        CaptionStyle captionStyle = systemCaptionStyle();
+
+        CaptioningManager manager = mock(CaptioningManager.class);
+        when(manager.getUserStyle()).thenReturn(captionStyle);
+        when(manager.getFontScale()).thenReturn(SYSTEM_TEXT_SCALE_RATIO);
+        when(manager.isEnabled()).thenReturn(testCase.captionsManagerIsEnabled);
+
+        Context context = mock(Context.class);
+        when(context.getSystemService(CAPTIONING_SERVICE)).thenReturn(manager);
+
+        SystemCaptionPreferences systemCaptionPreferences = new SystemCaptionPreferences(context);
+        systemCaptionPreferences.setAccessibilityCaptionsStyleEnabled(testCase.captionsLocallyEnabled);
+        return systemCaptionPreferences;
+    }
+
+    private static CaptionStyle systemCaptionStyle() {
+        try {
+            CaptionStyle style = mock(CaptionStyle.class);
+            when(style.hasForegroundColor()).thenReturn(true);
+            setFinalField(style, "foregroundColor", SYSTEM_FOREGROUND_COLOR);
+            when(style.hasBackgroundColor()).thenReturn(true);
+            setFinalField(style, "backgroundColor", SYSTEM_BACKGROUND_COLOR);
+            when(style.hasWindowColor()).thenReturn(true);
+            setFinalField(style, "windowColor", SYSTEM_WINDOW_COLOR);
+            when(style.hasEdgeColor()).thenReturn(true);
+            setFinalField(style, "edgeType", SYSTEM_EDGE_TYPE);
+            when(style.hasEdgeType()).thenReturn(true);
+            setFinalField(style, "edgeColor", SYSTEM_EDGE_COLOR);
+            when(style.getTypeface()).thenReturn(SYSTEM_TYPEFACE);
+            return style;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+}
```

Comment:
`No new line at the end of file`

Reply:
https://github.com/novoda/no-player/pull/277/commits/01eb575af6074460e4e7cdfdad1f73dcc0b78ea5